lib/src/main/java/com/shopify/checkoutsheetkit/BaseWebView.kt

```diff
@@ -0,0 +1,189 @@
+            override fun onProgressChanged(view: WebView?, newProgress: Int) {
+                super.onProgressChanged(view, newProgress)
+                getEventProcessor().updateProgressBar(newProgress)
+            }
+        }
+        isHorizontalScrollBarEnabled = false
+        requestDisallowInterceptTouchEvent(true)
+        setBackgroundColor(TRANSPARENT)
+        layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
+        id = View.generateViewId()
+    }
+
+    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
+        if (keyCode == KeyEvent.KEYCODE_BACK && canGoBack()) {
+            goBack()
+            return true
+        }
+        return super.onKeyDown(keyCode, event)
+    }
+    open inner class BaseWebViewClient : WebViewClient() {
+        init {
+            if (BuildConfig.DEBUG) {
+                setWebContentsDebuggingEnabled(true)
+            }
+        }
+
+        override fun onRenderProcessGone(view: WebView, detail: RenderProcessGoneDetail): Boolean {
+            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && !detail.didCrash()) {
+                // Renderer was killed because system ran out of memory.
+
+                val eventProcessor = getEventProcessor()
+                eventProcessor.onCheckoutViewFailedWithError(
+                    CheckoutSheetKitException(
+                        errorDescription = "Render process gone.",
+                        errorCode = CheckoutSheetKitException.RENDER_PROCESS_GONE,
+                        isRecoverable = recoverErrors,
+                    )
+                )
+                true
+            } else {
+                false
+            }
+        }
+
+        override fun onReceivedError(
+            view: WebView?,
+            request: WebResourceRequest?,
+            error: WebResourceError?
+        ) {
+            super.onReceivedError(view, request, error)
+            if (error != null) {
+                handleError(
+                    request,
+                    error.errorCode,
+                    error.description.toString()
+                )
+            }
+        }
+
+        override fun onReceivedHttpError(
+            view: WebView?,
+            request: WebResourceRequest?,
+            errorResponse: WebResourceResponse?
+        ) {
+            super.onReceivedHttpError(view, request, errorResponse)
+            if (errorResponse != null) {
+                handleError(
+                    request,
+                    errorResponse.statusCode,
+                    errorResponse.reasonPhrase,
+                    errorResponse.responseHeaders,
+                )
+            }
+        }
+
+        internal open fun isRecoverable(statusCode: Int): Boolean {
+            return when (statusCode) {
+                TOO_MANY_REQUESTS -> recoverErrors
+                in CLIENT_ERROR -> false
+                else -> recoverErrors
+            }
+        }
```

Comment:
It'd be >= 400 OR < 0.

< 0 would be related to the errors listed here (mostly around connecting to servers) https://developer.android.com/reference/android/webkit/WebViewClient#constants_1

Things like `ERROR_BAD_URL` aren't actually recoverable.  It might be worth adding a few more conditions here